test/unit/domain_test.rb

```diff
@@ -20,6 +20,14 @@ def setup
     assert !other_domain.save
   end
 
+  test "should have a unique case-insensitive name" do
+    other_domain = Domain.new(:name => @domain.name.upcase)
+    refute other_domain.save
+    assert_raises ActiveRecord::RecordNotUnique, ActiveRecord::StatementInvalid do
```

Comment:
The weirdest part is that if I remove the migration and add the index manually running `CREATE UNIQUE INDEX domain_name_index ON domains (name collate nocase)`, this test pass. The migration is certainly running for pg and sqlite as you can see on Jenkins console.